### PR TITLE
Clean up '#if CORE' in ProcessCodeMethods.cs

### PR DESCRIPTION
**Pull request by iSazonov**
Related #3565 and #4357.

### CODE REVIEW

File: src/System.Management.Automation/engine/ProcessCodeMethods.cs [hunk 1]
```diff
@@ -58,11 +58,8 @@ internal static int GetParentPid(Process process)
             Diagnostics.Assert(process != null, "Ensure process is not null before calling");
             PROCESS_BASIC_INFORMATION pbi;
             int size;
-#if CORECLR
             var res = NtQueryInformationProcess(process.SafeHandle.DangerousGetHandle(), 0, out pbi, Marshal.SizeOf<PROCESS_BASIC_INFORMATION>(), out size);
-#else
-            var res = NtQueryInformationProcess(process.Handle, 0, out pbi, Marshal.SizeOf<PROCESS_BASIC_INFORMATION>(), out size);
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/ProcessCodeMethods.cs]: `Process.Handle` is back, so probably we should use `NtQueryInformationProcess(process.Handle, ...` here.

Comment (iSazonov) [inline src/System.Management.Automation/engine/ProcessCodeMethods.cs]: It seems we should return in ClrFacade.cs `internal static SafeHandle GetSafeProcessHandle(Process process)` - we removed '#if CORE' there too.

Comment (daxian-dbw) [inline src/System.Management.Automation/engine/ProcessCodeMethods.cs]: That method in ClrFacade.cs should go away now I think.

Comment (iSazonov) [inline src/System.Management.Automation/engine/ProcessCodeMethods.cs]: Fixed.

**Review by daxian-dbw — APPROVED** [review]
LGTM

**Comment by iSazonov** [comment]
@daxian-dbw Could you please merge?

**Pull request merged**